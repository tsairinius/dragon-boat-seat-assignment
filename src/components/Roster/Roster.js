import React from 'react';
import Paddler from '../Paddler/Paddler';
import PropTypes from 'prop-types';

function Roster(props) {

    const paddlerComponents = props.paddlers.map(paddler => <Paddler key={paddler.id} paddlerProfile={paddler} handlePaddlerClick={props.handlePaddlerClick}/>)
    
    return (
        <div>
            <h1 style={{color: 'black', fontFamily: 'Allura', fontSize: '40px'}}>Paddlers</h1>
            {paddlerComponents}
        </div>
    );
};

Roster.propTypes = {
    label: PropTypes.string.isRequired,
    paddlers: PropTypes.array.isRequired,
    handlePaddlerClick: PropTypes.func.isRequired
};


export default Roster;