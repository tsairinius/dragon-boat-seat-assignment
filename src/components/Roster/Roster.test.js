import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Roster from './Roster';

afterEach(cleanup);

it('renders Roster component without crashing', () => {
  const container = document.createElement('div');
  ReactDOM.render(<Roster />, container);
});

it('rendered Roster matches snapshot', () => {
    const tree = renderer.create(<Roster />).toJSON();
    expect(tree).toMatchSnapshot();
});
