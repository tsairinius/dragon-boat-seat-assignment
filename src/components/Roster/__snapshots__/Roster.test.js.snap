// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered Roster matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid purple",
    }
  }
>
  <h1
    style={
      Object {
        "color": "black",
        "fontFamily": "Allura",
        "fontSize": "40px",
      }
    }
  >
    Paddlers
  </h1>
  <div
    style={
      Object {
        "border": "1px solid green",
      }
    }
  >
    Eric
  </div>
</div>
`;
